src/files-card-body.jsx

```diff
@@ -42,42 +42,23 @@ import "./files-card-body.scss";
 const _ = cockpit.gettext;
 
 const compare = (sortBy) => {
-    const compareFileType = (a, b) => {
-        const aIsDir = (a.type === "dir" || a?.to === "dir");
-        const bIsDir = (b.type === "dir" || b?.to === "dir");
-
-        if (aIsDir && !bIsDir)
-            return -1;
-        if (!aIsDir && bIsDir)
-            return 1;
-        return 0;
+    const dir_sort = (a, b) => {
+        return Number(b.to === "dir") - Number(a.to === "dir");
```

Comment:
I've stared at this for a while, and it feels like the wrong way around? If `b` is a dir and `a` is a file, then this would return 1, meaning that b > a, meaning that the file a sorts before dir b.

Can we have a test for the sort order?

Reply:
I also originally had it the other way.  I reversed it after manually testing it.

It could use a test in any case...

Reply:
Sorry, I was wrong -- it returns 1, but that means that a > b.